xwiki-pro-macros-api/src/main/resources/js/button-macro.js

```diff
@@ -0,0 +1,38 @@
+require(['jquery'], function($) {
+
+  let shouldAttachPicker = true;
+  let pickerRequired = true;
+
+  const attachPicker = function(config) {
+    shouldAttachPicker = false;
+    var iconOptions = {};
+    iconOptions['prefix'] = config.prefix;
+    $('.buttonIconPicker').xwikiIconPicker(iconOptions);
+  };
+  const requireIconPicker = function(config) {
+    pickerRequired = false;
+    require([config['icon-picker']], function () {
+      attachPicker(config);
+    });
+  };
+
+
+
+  let button = $('.buttonIconPicker');
+  if (button.length > 0) {
+    requireIconPicker(button.data('config'));
+  }
```

Comment:
This is not good. You initialize all icon pickers using the configuration from the first one. It's not a problem for now because all have the same configuration currently, but as soon as you add some differences you'll get into problems. Best is to do something like this:
```
$('.elementToEnhance').each(function() {
  enhance($(this), $(this).data('config'));
});
```